### PR TITLE
 [DNM] Stabsinthe!

### PR DESCRIPTION
**Pull request by cebutris**
:cl: Cebutris & CameronWoof
add: Stabsinthe! A new alcohol you can brew with absinthe and acid spit, or make an entire bottle by tablecrafting a full absinthe bottle with a nettle branch!
imgadd: Stabsinthe bottle icon by CameronWoof!
/:cl:

In the Spirit of the current Bastion drinks (Fetching Fizz and Hearty Punch), I wanted to add some more! Stabsinthe seemed pretty simple, so I wanted to try my hand at it. I gave it a unique recipe and even it's own tablecrafting recipe!
 
The idea of it is that it should cause people to attack people that touch them automatically, and force them into harm intent, sort of like a reverse Pax.

Sprites are still needed for the bottle, and maybe the glass

TODO:

- [x] framework
- [x] sprites
- [x] retaliation code
- [x] bugfixes

### CODE REVIEW

File: modular_citadel/code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/datum/crafting_recipe/food/stabsinthe //You can make a bottle of stabsinthe with a full bottle of absinthe and a nettle
```
Comment (cebutris) [inline modular_citadel/code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm:2]: The tablecrafting is a bit weird. It doesn't consume any of the required items, just spawning the result. It DOES move the absinthe from whatever container it was in into the new bottle (which already has 100 stabsinthe, resulting in 200u total). I have no idea why it happens like that!

**Comment by cebutris** [comment]
An unintended (but hilarious) side effect of this is that if you don't drink it fast enough you'll splash yourself with the glass or smash yourself with the bottle when you get forced into harm intent. I'm calling it a feature

**Comment by cebutris** [comment]
Update so it's not stale. It still isn't working anywhere close to how I want it to be but there's an idea for framework in place. Feedback or suggestions would be appreciated!

File: modular_citadel/code/modules/reagents/chemistry/reagents.dm [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/datum/reagent/proc/on_mob_attacked(mob/living/M) //Handler for reagent effects that occur when the mob with it inside it is attacked
+	return
+
+/datum/reagent/water/on_mob_attacked(mob/living/M)
```
Comment (cebutris) [inline modular_citadel/code/modules/reagents/chemistry/reagents.dm]: ignore this, this is just what I use for testing the proc. It'll be removed once I get things running proper

**Comment by cebutris** [comment]
This is ready for a review! There's still the tablecrafting bug that I need to sort out, but other than that this is in a good place for a review

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 3]
```diff
@@ -76,6 +76,9 @@
 	if(!affecting) //missing limb? we select the first bodypart (you can never have zero, because of chest)
 		affecting = bodyparts[1]
 	send_item_attack_message(I, user, affecting.name)
+	if(src.reagents) //CIT CHANGE. Handles datum/reagent/proc/on_mob_attacked
```
Comment (deathride58) [inline code/modules/mob/living/carbon/carbon_defense.dm]: No need for the src. here

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 4]
```diff
@@ -112,6 +112,9 @@
 	return 0
 
 /mob/living/carbon/human/proc/check_block()
+	if(src.reagents) //CIT CHANGE. Handles datum/reagent/proc/on_mob_attacked
```
Comment (deathride58) [inline code/modules/mob/living/carbon/human/human_defense.dm]: Ditto. No need for the src. here

**Review by deathride58 — CHANGES_REQUESTED** [review]

**Comment by cebutris** [comment]
Okay so this is a problem. I can't have the icon in a modular file because that would mess up the broken bottle sprite, but I also can't put it in the normal file because that will make merge conflicts...

**Comment by cebutris** [comment]
THERE'S NO MORE MERGE CONFLICT, MR.BOT, YOU CAN REMOVE THE TAG NOW

**Comment by cebutris** [comment]
So I tried to pull to fix the merge conflicts and I succeeded in breaking absolutley everything about my local repo. The only bug present with the PR as it stands is that the bottle lacks a broken sprite, and because I cocked everything up I don't think I can fix it in this PR. I CAN, however, make a followup PR to fix it. I'm just bad at Git.

I commented out the tablecraft recipe because I could not figure out WHY it wasn't working. It was as far as I could see identical to how other tablecraft recipes are handled, but it didn't behave the same way at all. Once someone figures out why it happened like that the recipe can be uncommented, I guess.

I'm considering this ready. As ready as it's going to get, in any case

**Comment by Poojawa** [comment]
since this was ported upstream, we don't need this anymore yah?

**Comment by cebutris** [comment]
Well, it might not be merged upstream. I was planning on closing the PR if the upstream one got merged, and if it didn't, working on fixing this one.

However I fucked my repo pretty hard so yeah closing it is fine

**Pull request closed without merging**